src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/AceEditor.java

```diff
@@ -3130,6 +3130,36 @@ private boolean rowEndsWithOpenBracket(int row)
    }
 
    @Override
+   public Range getParagraph(Position pos, int startRowLimit, int endRowLimit)
+   {
+      // assume start, end at current position
+      int startRow = pos.getRow();
+      int endRow   = pos.getRow();
+      
+      // walk backwards to beginning of paragraph
+      for (int i = pos.getRow() - 1; i >= startRowLimit; i--)
+      {
+         if (getLine(i).trim().isEmpty())
+         {
+            startRow = i + 1;
```

Comment:
Does this do the right thing if the paragraph is at the very start of the document? (Seems like `i = 0` wouldn't happen as `startRowLimit` will never be negative)